delivery_line_sale_line_position/report/report_deliveryslip.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+  <template id="report_delivery_document" inherit_id="stock.report_delivery_document">
+
+    <!-- Changes for when the picking is not yet done -->
+
+    <xpath expr="//table[@name='stock_move_table']/thead" position="before">
+      <t
+                t-set="has_line_position"
+                t-value="any(o.move_lines.filtered(lambda x: x.position_sale_line and x.product_uom_qty))"
+            />
+    </xpath>
+
+    <xpath expr="//th[@name='th_sm_product']" position="before">
+      <th t-if="has_line_position" name="th_sm_pos"><strong>Pos</strong></th>
+    </xpath>
+    <!-- The star in the selector may seem not very specific, but it is done -->
+    <!-- in purpose so it works with some other module that change this report. -->
+    <!-- Looking at you stock_picking_group_by_partner_by_carrier -->
+    <xpath expr="//table[@name='stock_move_table']/tbody/tr/*[1]" position="before">
+      <td t-if="has_line_position">
+        <span t-field="move.position_sale_line" />
+      </td>
+    </xpath>
+
+
+    <!-- Changes for when the picking is done -->
+
+    <xpath expr="//table[@name='stock_move_line_table']/thead" position="before">
+      <t
```

Comment:
do you really need to compute it twice?

Reply:
Yes, one computation for when the picking is in state done, and one for other state of the picking.
It is actually not the same computation